pallets/auctions/src/types/english.rs

```diff
@@ -0,0 +1,118 @@
+use super::*;
+
+///
+/// Implementation of EnglishAuction
+///
+impl<T: Config> NftAuction<T::AccountId, T::AuctionId, BalanceOf<T>, Auction<T>, Bid<T>> for EnglishAuction<T> {
+	fn create(&self, sender: T::AccountId, auction: &Auction<T>) -> DispatchResult {
+		self.validate_data()?;
+		Pallet::<T>::validate_create(&self.common_data)?;
+		Pallet::<T>::handle_create(sender, auction, &self.common_data)?;
+
+		Ok(())
+	}
+
+	fn update(self, sender: T::AccountId, auction_id: T::AuctionId) -> DispatchResult {
+		self.validate_data()?;
+
+		<Auctions<T>>::try_mutate(auction_id, |maybe_auction| -> DispatchResult {
+			let auction_result = maybe_auction.as_mut().ok_or(Error::<T>::AuctionDoesNotExist)?;
+
+			if let Auction::English(english_auction) = auction_result {
+				Pallet::<T>::validate_update(sender, &english_auction.common_data)?;
+				*english_auction = self;
+
+				Ok(())
+			} else {
+				Err(Error::<T>::NoChangeOfAuctionType.into())
+			}
+		})
+	}
+
+	///
+	/// Places a bid on an EnglishAuction
+	///
+	/// - removes lock on auction.common_data.last_bid
+	/// - sets lock on new bid
+	/// - updates auction.common_data.last_bid and auction.common_data.next_bid_min
+	/// - if necessary, increases auction end time to prevent sniping
+	///
+	fn bid(&mut self, _auction_id: T::AuctionId, bidder: T::AccountId, bid: &Bid<T>) -> DispatchResult {
```

Comment:
This and a bunch of other functions should probably be `require_transactional`

Reply:
Good catch thx (that's my first pallet so i'm still learning)